src/nsyt/remote.rs

```diff
@@ -56,19 +44,16 @@ async fn process_results(
     output_path: &str,
 ) -> i32 {
     let mut valid_img_count = 0;
-    for image in results.items.iter() {
-        // If this image is a thumbnail and we're ignoring those, then ignore it.
-        if image.is_thumbnail && !thumbnails {
-            continue;
-        }
-
-        // If we're searching for a substring and this image doesn't match, skip it.
-        if !search_empty_or_has_match(&image.imageid, search) {
-            continue;
-        }
-        valid_img_count += 1; //ITM is an anti-pattern. TODO: enumerate(), and have the 'e' fall out.
+    for (idx, image) in results
+        .items
+        .iter()
+        .filter(|image| {
+            image.is_thumbnail && !thumbnails && search.iter().any(|i| image.imageid.contains(i))
+        })
+        .enumerate()
+    {
+        valid_img_count = idx as i32; //ITM is an anti-pattern. TODO: enumerate(), and have the 'e' fall out.
```

Comment:
what you want for the valid img count var is the length of the iterator

Reply:
Any changes made here can _probably_ be made to `src/m20/remote.rs` and `src/msl/remote.rs`

This does look better than previous.